### PR TITLE
Replace "star import" with "import as" statement

### LINKED ISSUE #49416: A few years after #701 and PyTorch is still using implicit __all__ imports.
> i dont think we'll be fixing this. (i dont know if there's a way to do it either).

_Originally posted by @soumith in https://github.com/pytorch/pytorch/issues/701#issuecomment-329026903_

> > Maybe one could change `from torch._C import *` to `from torch._C import from_numpy, ...`?
> This seems simple enough. Are we certain this is the way to go? 
> I thought we'd need type hints (which would be way harder to convince people to use).
> Should we open another issue to discuss @moi90's solution?

_Originally posted by @cardoso-neto in https://github.com/pytorch/pytorch/issues/701#issuecomment-485272641_

A quick search for `import *` yields hundreds results.

If requested by a team member, I'd gladly do the tedious work replacing the import statements and pull requesting. :)
I would probably just need some pointers on how to do find out what is being imported from which C module, but I'd make quick work of it.
Long import statements could be indented very cleanly following [psf/black's](https://github.com/psf/black) style guides.

### PR DESCRIPTION
**Pull request by cardoso-neto**
As part of a larger effort to remediate #49416 (it'll take more than one PR to fix it):

On `torch/utils/hipify/cuda_to_hip_mappings.py` I decided to go with an `import module as m` statement instead of the usual `from module import ...`.
The scenario:
[removestar](https://github.com/asmeurer/removestar) changed `from .constants import *` to
```python
from .constants import (
    API_BLAS, API_C10, API_CAFFE2, API_DRIVER, API_FFT, API_PYTORCH, API_RAND,
    # many lines of this
    ...
    CONV_THREAD, CONV_TYPE, CONV_VDPAU, CONV_VERSION, HIP_UNSUPPORTED,
)
```
which creates a whole mess and adds another place to edit when adding/renaming/removing a constant.
I used this OR regex `(API_BLAS|API_C10|API_CAFFE2|API_DRIVER|...)` to replace everything with `c.$0`.
And now the import just reads
```python
from . import constants as c
```

### CODE REVIEW

**Comment by cardoso-neto** [comment]
I just signed the Contributor License Agreement. Ping me if there's anything else.

**Comment by cardoso-neto** [comment]
The failed checks don't seem related to the changes I've made. 🤔

**Comment by mruberry** [comment]
@ngimel, are you the right person to review this?

**Comment by ngimel** [comment]
No, someone from AMD is.

**Comment by jeffdaily** [comment]
@cardoso-neto what was wrong with your first approach?  How many lines were added by using `from module import ...`, as indicated by your comment `# many lines of this`.

This mappings file is updated infrequently, and the set "constants" in the file, e.g., API_BLAS, API_C10, are not changing.  The only changes we make, if any, is to the content of such constants rather than adding new ones.  In short, I don't understand how this approach adds to the maintenance burden?

**Comment by jeffdaily** [comment]
I tried removestar on cuda_to_hip_mappings.py and got this:

```
--- original/cuda_to_hip_mappings.py
+++ fixed/cuda_to_hip_mappings.py
@@ -1,6 +1,13 @@
 import collections

-from .constants import *
+from .constants import (API_BLAS, API_C10, API_CAFFE2, API_DRIVER, API_FFT, API_PYTORCH, API_RAND,
+                        API_ROCTX, API_RTC, API_RUNTIME, API_SPARSE, CONV_CACHE, CONV_CONTEXT,
+                        CONV_D3D10, CONV_D3D11, CONV_D3D9, CONV_DEF, CONV_DEVICE, CONV_DEVICE_FUNC,
+                        CONV_EGL, CONV_ERROR, CONV_EVENT, CONV_EXEC, CONV_GL, CONV_GRAPHICS,
+                        CONV_INCLUDE, CONV_INCLUDE_CUDA_MAIN_H, CONV_INIT, CONV_JIT, CONV_MATH_FUNC,
+                        CONV_MEM, CONV_MODULE, CONV_NUMERIC_LITERAL, CONV_OCCUPANCY, CONV_OTHER,
+                        CONV_PEER, CONV_SPECIAL_FUNC, CONV_STREAM, CONV_SURFACE, CONV_TEX,
+                        CONV_THREAD, CONV_TYPE, CONV_VDPAU, CONV_VERSION, HIP_UNSUPPORTED)

 """ Mapping of CUDA functions, include files, constants, and types to ROCm/HIP equivalents
 This closely follows the implementation in hipify-clang
```

This seems fine to me, compared to the large diff proposed by the current PR.

**Comment by cardoso-neto** [comment]
@jeffdaily 
>what was wrong with your first approach

Wrong is a strong word, but it was inferior to my latter approach, mainly because it introduced another place to maintain whenever adding, renaming, or removing a constant.
And, if the dev making that change were not using an [automatic import sorter](https://github.com/PyCQA/isort), he would need to fix the multi-line import manually.

>The only changes we make, if any, is to the content of such constants

Well, yeah, in that very specific case, you got me.
But as a general rule for situations like this (accessing many constants from the same module), using `import module as m` is the style that makes more sense.
And though I could argue why in the general case by telling you
- it doesn't pollute the name space;
- it's more readable in the code because you know what comes from where,
and that sort of thing, the only argument I have for this specific case is that it is good practice to push for standardization.

Especially when **the only con** we have is that it created a large diff.

@ezyang 
>kind of weird file to attempt first

It's just the "biggest offender" and knowing that no one likes _large diffs_, I thought I'd do this one separately. But I already have another branch in which I'm tackling the `__init__.py` which you can find linked on that issue.

>it isn't public API

Indeed it isn't, but I don't think that that should allow it to get away with a wildcard import.


In summary, I strongly believe my PR is for the best of the project.
However, if you guys still don't see things my way just tell me and I'll force push changes similar to what jeffdaily said seems fine to him (I'll run `isort` with multi line output mode 5 on `removestar`'s output and it'll look like the below example).

```python
from .constants import (
    API_BLAS, API_C10, API_CAFFE2,  # all the way to the character limit of 88 ...
    ...  # 5-8 lines of this
    HIP_UNSUPPORTED,  # trailing comma
)  # dedented closing bracket 
```
Line length source: [black's standard](https://github.com/psf/black/blob/master/docs/the_black_code_style.md#line-length)
Other import styles: [isort's settings](https://pycqa.github.io/isort/#multi-line-output-modes)

**Comment by cardoso-neto** [comment]
Just for the sake of reproducibility:
The full OR regex I used was

```(API_BLAS|API_C10|API_CAFFE2|API_DRIVER|API_FFT|API_PYTORCH|API_RAND|API_ROCTX|API_RTC|API_RUNTIME|API_SPARSE|CONV_CACHE|CONV_CONTEXT|CONV_D3D10|CONV_D3D11|CONV_D3D9|CONV_DEF|CONV_DEVICE|CONV_DEVICE_FUNC|CONV_EGL|CONV_ERROR|CONV_EVENT|CONV_EXEC|CONV_GL|CONV_GRAPHICS|CONV_INCLUDE|CONV_INCLUDE_CUDA_MAIN_H|CONV_INIT|CONV_JIT|CONV_MATH_FUNC|CONV_MEM|CONV_MODULE|CONV_NUMERIC_LITERAL|CONV_OCCUPANCY|CONV_OTHER|CONV_PEER|CONV_SPECIAL_FUNC|CONV_STREAM|CONV_SURFACE|CONV_TEX|CONV_THREAD|CONV_TYPE|CONV_VDPAU|CONV_VERSION|HIP_UNSUPPORTED)```

And I obtained it by running removestar on `cuda_to_hip_mappings.py`, copying its line 3 to a python string and running:
```python
string.replace(",", "|").replace(" ", "").replace("\n", "")
```
Then on VS Code find & replace, I pasted the OR regex on find and `c.$0` on replace.

**Comment by rgommers** [comment]
I agree with @jeffdaily here, the `removestar` approach is nicer, and this can be a diff of a few lines rather than thousands of lines. Other modules should use `from xxx import name1, name2, ...` as well rather than this approach.
